### PR TITLE
Service for Excel file generation

### PR DESCRIPTION
**Pull request by FrancisRenaud**
The added service receives an object from the frontend and streams an Excel file back

### CODE REVIEW

**Comment by FrancisRenaud** [comment]
@latch2112 Thanks for the review. Do we have documentation on the process of updating the api?

**Comment by latch2112** [comment]
every folder under api is a resource
when we build the service we split it per resource at service level (each resource became it's own service in term of deployment and uses it's own port)
the file docs.json under each folder is its "API documentation" using OpenApi 3.0 
i.e. /api/variant/docs
please, find the proper route to add the excel sub-resource (do you generate an excel table of the resource patient, variant, gene or meta ?)
Or if excel is a resource in itself, we need to create a new service for it (let me know, I'll try to provide more guideline as to create a new service (this will a bit more complex and now we need to involve Eric (devops) to let him know about this new service

**Comment by latch2112** [comment]
Sorry the lack of proper/better documentation, we build the entire projet with a team of 2-3 devs only and due to human resource constraint we have to cut corner somewhere (like in the development standard)
We also tried to use the most simple code possible and auto-document it thru the usage of package.json script and a bogus service named boilerplate to follow as a standard

**Comment by latch2112** [comment]
and last be not least, service.js (where you put your route) is clearly a 'generic service loader' that need configuration to start different services

**Comment by FrancisRenaud** [comment]
@latch2112 Thanks for the clarifications. I was looking for something like a Swagger file and wasn't sure if the doc was auto-generated of manually written. In the las few years I was mostly using GraphQL instead of Swagger and the doc got auto-generated from the annotations on my resolvers etc. I didn't think too much about looking at the boilerplate service. All the above comments will help!

**Comment by FrancisRenaud** [comment]
The endpoint code will be moved under variant as this is where it's relevant for now.

**Comment by Magnitus-** [comment]
@FrancisRenaud For the merge conflicts with the apply.sh file, just resolve it by removing the file from your branch. The file was removed from the repo.

File: apply.sh [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export DEFAULT_PROD_VERSION=1.4.0
+export DEFAULT_PROD_VERSION=1.5.0
```
Comment (Magnitus-) [inline apply.sh:1]: Just remove this file from your branch to resolve the merge conflict after rebasing/merging.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "clin-proxy-api",
-  "version": "1.4.0",
+  "version": "1.5.0",
```
Comment (Magnitus-) [inline package.json:3]: Still good. Version in the dev was bumped to 1.4.1. Your version should still be 1.5.0.

File: src/services/api/variant/index.js [hunk 3]
```diff
@@ -180,6 +181,29 @@ export default class VariantService extends ApiService {
 
         } ) )
 
+        // Generate excel file for list of variants
+        this.instance.post( {
+          path: `${this.config.endpoint}/xl`,
+        }, async (req, res, next) => {
```
Comment (Magnitus-) [inline src/services/api/variant/index.js:187]: I find wrapping sendDataAsExcel in an endpoint handler makes things more convoluted then they have to be.

I understand the desired to decouple error login from the endpoint handler though you do have some choices here:
- Make  sendDataAsExcel a higher order function and either pass the logger or even complete error handler to it as an argument. An example of this pattern can be found here (also, you might take the opportunity to log your error message in json format which will make it easier to centralize later): https://github.com/cr-ste-justine/score-auth/blob/master/src/middleware/access_control.js#L21
- Put a separate error handler that logs the message (you can do it with Express and I believe Restify is pretty similar in behavior)

Comment (Magnitus-) [reply src/services/api/variant/index.js:187]: After looking at the remainder of the code, I see that it follows a similar pattern.

While I think the implementation adds extra complexity without giving value in return, I will not force my opinion on a single new endpoint without at least refactoring other endpoints in the same file which is out of scope for this PR I so I'll close this discussion and leave it be.

**Pull request merged**